src/gc/gc.cpp

```diff
@@ -1590,13 +1587,10 @@ inline bool can_use_write_watch_for_card_table()
 void WaitLongerNoInstru (int i)
 {
     // every 8th attempt:
-    Thread *pCurThread = GCToEEInterface::GetThread();
-    bool bToggleGC = false;
-    if (pCurThread)
-    {
-        bToggleGC = GCToEEInterface::IsPreemptiveGCDisabled(pCurThread);
-        if (bToggleGC)
-            GCToEEInterface::EnablePreemptiveGC(pCurThread);
+    Thread* pCurThread = GCToEEInterface::GetThread();
+    bool bToggleGC = GCToEEInterface::IsPreemptiveGCDisabled();
```

Comment:
There was null check for `pCurThread`. Shouldn't here be `assert(pCurThread!= nullptr);` ?

Reply:
That wouldn't be the same semantically. There is a branch on `pCurThread` being null later on in the function, so it wouldn't be correct to assert that it's not null here (it may very well be).

The null check was due to the contract of `EnablePreemptiveGC` - namely that you can't pass in a null pointer to it.